url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ComboCall.java

```diff
@@ -0,0 +1,350 @@
+package net.nexon.vindictus.itemcomparer.logic;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import net.nexon.vindictus.itemcomparer.modell.Combo;
+import net.nexon.vindictus.itemcomparer.modell.ext.Armor;
+import net.nexon.vindictus.itemcomparer.modell.ext.Gloves;
+import net.nexon.vindictus.itemcomparer.modell.ext.Helm;
+import net.nexon.vindictus.itemcomparer.modell.ext.Pants;
+import net.nexon.vindictus.itemcomparer.modell.ext.Shoes;
+
+public class ComboCall implements Callable<List<Combo>> {
+
+	private List<Shoes> shoes;
+	private List<Pants> pants;
+	private List<Gloves> gloves;
+	private List<Armor> armors;
+	private List<Helm> helms;
+
+	private Comparator<Combo> comp;
+
+	private int results = 30;
+	private double PRICE;
+	private int ATK;
+	private int MATK;
+	private int[] MAXIS;
+
+	public ComboCall(double pri, int atk, int matk, int[] maxima,
+			List<Shoes> sho, List<Pants> pan, List<Gloves> glov,
+			List<Armor> arms, List<Helm> hes, int result, Comparator<Combo> defc) {
+		PRICE = pri;
+		shoes = sho;
+		pants = pan;
+		gloves = glov;
+		armors = arms;
+		helms = hes;
+		results = result;
+		comp = defc;
+		MAXIS = maxima;
+		ATK = atk;
+		MATK = matk;
+		thresh = results * 2;
+	}
+
+	private static int min = -1;
+
+	private List<Combo> combos = new ArrayList<>();
+
+	private double total_p = 0;
+	private double s_p = 0;
+	private double p_p = 0;
+	private double g_p = 0;
+	private double a_p = 0;
+	private double h_p = 0;
+	private int thresh;
+
+	@Override
+	public List<Combo> call() throws Exception {
+
+		String compname = comp.getClass().getSimpleName();
+
+		switch (compname) {
+		case "AtkComparator":
+			return callAtk();
+		case "DefComparator":
+			return callDef();
+		case "MatkComparator":
+			return callMatk();
+		case "StaDComparator":
+		case "StaAComparator":
+		case "StaMComparator":
+			return callSta();
+		default:
+			return callStandard();
+		}
+	}
+
+	private List<Combo> callStandard() throws Exception {
+		int i = 0;
+		for (Shoes s : shoes) {
+			s_p = s.getTotalPrice();
+			for (Pants p : pants) {
+				p_p = p.getTotalPrice();
+				for (Gloves g : gloves) {
+					g_p = g.getTotalPrice();
+					for (Armor a : armors) {
+						a_p = a.getTotalPrice();
+						for (Helm h : helms) {
+							h_p = h.getTotalPrice();
+							total_p = s_p + p_p + g_p + a_p + h_p;
+
+							if (total_p <= PRICE) {
+								combos.add(new Combo(s, p, g, a, h));
+							}
+							i++;
+							if (i % 1000000 == 0) {
+								System.out.print(".");
+							}
+							if (combos.size() > thresh) {
+								Collections.sort(combos, comp);
+								combos = new ArrayList<Combo>(combos.subList(0,
+										results));
+							}
+						}
+					}
+				}
+			}
+		}
+		System.out.print("x");
+		return combos;
+
+	}
+
+	private List<Combo> callDef() throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.